### DISCUSSION TITLE
Is this how prenote works ? and Prenote Process

### DISCUSSION THREAD

**Discussion by samyje** [moov-io/ach#1006] [Q&A] [answered]
Hi, 

With all your help we are nearing the completion for verifying the bank details using NACHA prenote. 

So here is the process we have implemented in our system. Please let me know if there is a mistake/suggestion. 

**Background:**
User adds the bank details in our system, and this needs to be verified because of the latest NACHA regulations (if payments are done using NACHA). It is a modern web-based application where a user can add/edit/delete the data added. Bank details are one of the data that gets added in our system. On a typical day, 50 to 1000s of bank details are getting added into our system. So, this Prenote verification needs to be automated.

**Process:**
On daily basis, we created a job to export these added bank details in NACHA format. These exported files are placed in FTP path that is connected with the bank.  There is also an import happening on daily basis if the bank sends any return files. We match the bank & routing number of the bank details and update the return code in the system(from the file that bank sends). If added bank detail has not received any return code or NOC from bank within 6 days from the date of adding, the system will automatically consider the added bank details as verified. 

In the 6 day time period user cannot update the bank details as it is sent for verification. System will allow the user to update the bank detail only if there is a return or NOC from bank for the added bank detail. 

**Why 6-day limit?**
From the research I have done, bank will take time to process the details ( 3-6 days). So I believe it is good to have the 6-day limit. 

**Why system will automatically consider the added bank detail as Prenote verified ?**
Bank will only send return code or NOC if there is something wrong in the added bank details. If there is nothing wrong in the details it is believed we will not receive any updates from the bank for the added  detail. So, the system will automatically consider the bank detail as verified.

**Can user edit/delete added Bank details any time ?**

No, once the details are added, user can only update the added details if there is a return code available ( which is from bank). After 6-days the bank detail is auto approved in our system so they cant edit or delete once it is added. If we let user edit as they wish same details will be exported everyday. 

Please let me know your suggestions or comments on this. Thanks a lot for your wonderful contribution.

**Comment by adamdecaf** [comment, marked as answer]
Yep! Your understanding of prenotes is how I understand them as well. 

> If there is nothing wrong in the details it is believed we will not receive any updates from the bank for the added detail. 

This is the most important line. It takes _days_ for the process to complete and only errors are returned.

**Reply by samyje** [reply]
Thank you @adamdecaf for your comments.
